### PR TITLE
BUG-116111 Register datalake validation improvement

### PR DESCRIPTION
**Pull request by schfeca75**
@lacikaaa or @keyki pls review

### CODE REVIEW

File: core/src/main/java/com/sequenceiq/cloudbreak/service/environment/EnvironmentService.java [hunk 1]
```diff
@@ -397,10 +408,10 @@ public DetailedEnvironmentResponse registerExternalDatalake(String environmentNa
                     Map<String, Map<String, String>> serviceSecretParamMap = StringUtils.isEmpty(registerDatalakeRequest.getRangerAdminPassword())
                             ? new HashMap<>() : Map.ofEntries(Map.entry(ServiceDescriptorDefinitionProvider.RANGER_SERVICE, Map.ofEntries(
                             Map.entry(ServiceDescriptorDefinitionProvider.RANGER_ADMIN_PWD_KEY, registerDatalakeRequest.getRangerAdminPassword()))));
-                    DatalakeResources datalakeResources = datalakeConfigProvider.collectAndStoreDatalakeResources(environmentName, datalakeAmbariUrl,
+                    DatalakeResources datalakeResources = datalakeConfigProvider.collectAndStoreDatalakeResources(environmentName, envView, datalakeAmbariUrl,
                             ambariUrl.getHost(), ambariUrl.getHost(), ambariClient, serviceSecretParamMap, ldapConfig, kerberosConfig, rdssConfigs,
                             environment.getWorkspace());
-                    environment.setDatalakeResources(datalakeResources);
+                    environment.setDatalakeResources(Sets.newHashSet(datalakeResources));
```
Comment (lacikaaa) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/environment/EnvironmentService.java]: shouldn't this be like: environment.getDatalakeResources().add(datalakeResources) ?

File: core/src/main/java/com/sequenceiq/cloudbreak/converter/environment/EnvironmentToDetailedEnvironmentResponseConverter.java [hunk 2]
```diff
@@ -90,8 +90,8 @@ public DetailedEnvironmentResponse convert(Environment source) {
                         .stream()
                         .map(StackViewResponse::getName)
                         .collect(Collectors.toSet()));
-        if (source.getDatalakeResources() != null) {
-            DatalakeResources datalakeResources = source.getDatalakeResources();
+        if (!CollectionUtils.isEmpty(source.getDatalakeResources())) {
+            DatalakeResources datalakeResources = source.getDatalakeResources().stream().findFirst().get();
```
Comment (lacikaaa) [inline core/src/main/java/com/sequenceiq/cloudbreak/converter/environment/EnvironmentToDetailedEnvironmentResponseConverter.java]: maybe the response should contain a set too

**Comment by keyki** [comment]
Can this be merged?

**Comment by lacikaaa** [comment]
LGTM, but it will require UI changes too I think

**Pull request merged**